src/librustdoc/core.rs

```diff
@@ -235,6 +236,24 @@ pub fn run_core(search_paths: SearchPaths,
         let mut sess = session::build_session_(
             sessopts, cpath, diagnostic_handler, codemap,
         );
+
+        lint::builtin::HardwiredLints.get_lints()
+                                     .into_iter()
+                                     .chain(rustc_lint::SoftLints.get_lints().into_iter())
+                                     .filter_map(|lint| {
+                                         if lint.name == warnings_lint_name ||
+                                            lint.name == intra_link_resolution_failure_name ||
+                                            lint.name == missing_docs {
+                                             None
```

Comment:
Took me a while to figure this out. So you are creating a list of *all* lints except for those you want to keep, and then you set all of them to `allow`.

Can you add a comment about this here?

Explanation about why I was confused:

I initially assumed we'd build a list of the lints we want to keep (so essentially `vec![warnings_lint_name, intra_link_resolution_failure_name, missing_docs]`, set `--cap-lints=allow` and then make sure that cap-lints doesn't override our specific set of lints.

r=me with comment